src/main/java/jsr223/docker/file/DockerFileScriptEngine.java

```diff
@@ -0,0 +1,164 @@
+
+    public DockerFileScriptEngine() {
+        // This is the entry-point of the script engine
+        log4jConfigurationLoader.loadLog4jConfiguration();
+    }
+
+    @Override
+    public Object eval(String script, ScriptContext context) throws ScriptException {
+
+        // Create docker file command - a simple docker build command 
+        String[] dockerFileCommand = dockerFileCommandCreator.createDockerFileExecutionCommand();
+
+        // Create a process builder
+        ProcessBuilder processBuilder = SingletonProcessBuilderFactory.getInstance()
+                                                                      .getProcessBuilder(dockerFileCommand);
+
+        // Use process builder environment and fill it with environment variables
+        Map<String, String> variablesMap = processBuilder.environment();
+
+        // Add string bindings as environment variables
+        stringBindingsAdder.addBindingToStringMap(context.getBindings(ScriptContext.ENGINE_SCOPE), variablesMap);
+
+        // Add DOCKER_HOST variable to execution environment
+        variablesMap.put(DOCKER_HOST_PROPERTY_NAME, DockerFilePropertyLoader.getInstance().getDockerHost());
+
+        // Replace variables in configuration file
+        String scriptReplacedVariables = variablesReplacer.replaceVariables(script, variablesMap);
+
+        File dockerfile = null;
+
+        Thread shutdownHook = null;
+
+        try {
+            dockerfile = configurationFileWriter.forceFileToDisk(scriptReplacedVariables,
+                                                                 dockerFileCommandCreator.FILENAME);
+
+            // Start process
+            Process process = processBuilder.start();
+
+            // Attach streams
+            processBuilderUtilities.attachStreamsToProcess(process,
+                                                           context.getWriter(),
+                                                           context.getErrorWriter(),
+                                                           context.getReader());
+
+            // Wait for process to exit
+            int exitValue = process.waitFor();
+
+            if (exitValue != 0) {
+                throw new ScriptException("Docker File failed with exit code " + exitValue);
+            }
+            return exitValue;
+        } catch (IOException e) {
+            log.warn("Failed to execute Docker File.", e);
+        } catch (InterruptedException e) {
+            log.info("Container execution interrupted. " + e.getMessage());
+        } finally {
+
+            // Delete configuration file
+            if (dockerfile != null) {
+                boolean deleted = dockerfile.delete();
+                if (!deleted) {
+                    log.warn("File: " + dockerfile.getAbsolutePath() + " was not deleted.");
+                }
+            }
+            if (shutdownHook != null) {
+                Runtime.getRuntime().removeShutdownHook(shutdownHook);
```

Comment:
I think we don't need  to handle a shutdown hook for the `docker build` command.

Reply:
From the docker documentation `If the Docker client loses connection to the daemon, the build is canceled. This happens if you interrupt the Docker client with CTRL-c or if the Docker client is killed for any reason. If the build initiated a pull which is still running at the time the build is cancelled, the pull is cancelled as well.` 

So it's a very good idea to kill/interrupt the client